### PR TITLE
Autocomplete Search

### PR DESCRIPTION
**Pull request by povodok**
#3

### CODE REVIEW

File: app/assets/stylesheets/autocomplete.css [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.autocomplete-suggestions { border: 1px solid #999; background: #FFF; overflow: auto; }
+.autocomplete-suggestion { padding: 2px 5px; white-space: nowrap; overflow: hidden; }
+.autocomplete-selected { background: #F0F0F0; }
+.autocomplete-suggestions strong { font-weight: normal; color: #3399FF; }
+.autocomplete-group { padding: 2px 5px; }
+.autocomplete-group strong { display: block; border-bottom: 1px solid #000; }
```
Comment (shhavel) [inline app/assets/stylesheets/autocomplete.css]: Please add empty line.

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class HomeController < ApplicationController
+  def autocomplete
+    @terms = Term.where("name LIKE ? AND entries_count > 0", "#{params[:query]}%").order('popularity DESC').limit(10)
+    render json: { suggestions: @terms.pluck(:name) }
+  end
```
Comment (shhavel) [inline app/controllers/home_controller.rb]: Also please add empty line.

File: public/css/autocomplete.css [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.autocomplete-suggestions { border: 1px solid #999; background: #FFF; overflow: auto; }
```
Comment (shhavel) [inline public/css/autocomplete.css]: This file should be removed.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -22,5 +22,7 @@
       </div><!--/row-->
     </div> <!-- /container -->
     <%= javascript_include_tag 'application' %>
+  <script src="/js/jquery.autocomplete.min.js"></script>
```
Comment (shhavel) [inline app/views/layouts/application.html.erb]: Add align (2 more whitespaces) same as on previous line.

**Review by shhavel — CHANGES_REQUESTED** [review]

**Review by shhavel — APPROVED** [review]

File: app/controllers/home_controller.rb [hunk 5]
```diff
@@ -1,7 +1,9 @@
 class HomeController < ApplicationController
   def autocomplete
-    render json: Term.where("name LIKE ? AND entries_count > ?" , "#{params[:query]}%", 1).order('popularity DESC').limit(10)
+    @terms = Term.where("name LIKE ? AND entries_count > 0", "#{params[:query]}%").order('popularity DESC').limit(10)
```
Comment (mahasamatman) [inline app/controllers/home_controller.rb]: Would be good to move logic to model, e.g. `@terms = Term.autocomplete_search(params[:query])`.

**Review by mahasamatman — CHANGES_REQUESTED** [review]

**Review by mahasamatman — APPROVED** [review]
Great work! So good so far.

**Pull request merged**